### ISSUE TITLE
Automatic limiting of number of running tasks based on memory

### ISSUE THREAD

**Issue by mathjeff** [gradle/gradle#18669]
### Current Behavior
Does Gradle have any interest in offering support to limit the number of running tasks based on available memory?

### Context
AndroidX discovered that lintAnalyze can take ~512M of memory and if we attempt to run ~60 of them in parallel then that might result in memory pressure and reduce the performance compared to running fewer of those tasks at a time.

Limiting the number of tasks based on available memory might also make it easier for users to notice if insufficient memory is slowing their builds down.

In the meanwhile we're looking into limiting the number of concurrent copies of lint via a shared BuildService which declares a limit on the number of concurrent usages

**Comment by ov7a** [comment]
Thank you for your interest in Gradle! Sorry for the late reply.

This issue will be closed as a duplicate of

* https://github.com/gradle/gradle/issues/20262

Please add your use case and 👍 to that issue. 

***

The other has a slightly better description and covers more cases.
